### ISSUE TITLE
Cron jobs are not working properly

### ISSUE THREAD

**Issue by ShaneSpinuzzi** [chelmertz/mantis-triggerclose#8]
I am not sure what is going on but the TriggerClose plugin no longer works when running as a cron job.

My email is full of messages with the output below.

How do I go about troubleshooting or fixing this?

Could not login with given user, check TriggerClose's plugin settings in the GUI

TriggerCloseApi.php [options]

Close Mantis plugins based on inactivity.

OPTIONS

	-v		Verbose output, print closed issues
	-h | --help	This helptext
	-n | --dry-run	Only print what would happen, do not change any statuses

**Comment by chelmertz** [comment]
I no longer have a system with mantis installed so I would not consider this plugin well maintained. However, I could try to help you by looking through the code.

What does the cron job look like? Try adding the -v flag for more verbosity in the error messages. It should be one of these failures: https://github.com/chelmertz/mantis-triggerclose/blob/master/TriggerClose/TriggerCloseApi.php#L120 so check your user too.

**Comment by ShaneSpinuzzi** [comment]
I found out that the cause to this issue is that the username.

The error I get with the option -v was: Could not login as user with saved ID

If the username has a space in it like "my username"  then the plugin fails to work.

Something I do not understand is the plugin calls  plugin_config_get('user'); which outputs the user id not the username. How does the plugin fail when there is a space in the username but the username is not used to login by the plugin? 

I did however manage to fix this by changing line 126 in the TriggerCloseApi.php file;

From: if(!auth_attempt_script_login(user_get_name($user_id))) {

To: if ( !auth_attempt_script_login( user_get_field( $user_id, 'username' )))  {

It works 100% now.

Thank You for the reply @chelmertz
